src/conbatant_repository.py

```diff
@@ -0,0 +1,26 @@
+from combatant import Combatant
+import requests
+
+
+class CombatantRepository:
+
+    @staticmethod
+    def get(name: str) -> Combatant:
+        data = requests.get(f"https://www.dnd5eapi.co/api/monsters/{name}")
+        if data.status_code != 200:
+            raise ValueError(f"Combatant with name [{name}] does not exist.")
+        json_data = data.json()
+
+        return CombatantRepository.__combatant_decoder(json_data)
```

Comment:
`@classmethod` instead of `@staticmethod`
and here you'll be able to user `cls.__combatant_decoder`